packages/core/src/auth/auth2.ts

```diff
@@ -0,0 +1,330 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as jose from 'jose'
+import {
+    GetSsoTokenParams,
+    getSsoTokenRequestType,
+    GetSsoTokenResult,
+    IamIdentityCenterSsoTokenSource,
+    InvalidateSsoTokenParams,
+    invalidateSsoTokenRequestType,
+    ProfileKind,
+    UpdateProfileParams,
+    updateProfileRequestType,
+    SsoTokenChangedParams,
+    ssoTokenChangedRequestType,
+    AwsBuilderIdSsoTokenSource,
+    UpdateCredentialsParams,
+    AwsErrorCodes,
+    SsoTokenSourceKind,
+    listProfilesRequestType,
+    ListProfilesResult,
+    UpdateProfileResult,
+    InvalidateSsoTokenResult,
+    AuthorizationFlowKind,
+    CancellationToken,
+    CancellationTokenSource,
+    bearerCredentialsDeleteNotificationType,
+    bearerCredentialsUpdateRequestType,
+    SsoTokenChangedKind,
+} from '@aws/language-server-runtimes/protocol'
+import { LanguageClient } from 'vscode-languageclient'
+import { getLogger } from '../shared/logger/logger'
+import { ToolkitError } from '../shared/errors'
+import { useDeviceFlow } from './sso/ssoAccessTokenProvider'
+
+export const AuthStates = {
+    NOT_CONNECTED: 'notConnected',
+    CONNECTED: 'connected',
+    EXPIRED: 'expired',
+} as const
```

Comment:
a map of type values can be useful if it's necessary to iterate at runtime, but otherwise isn't needed.